helix-term/src/application.rs

```diff
@@ -42,8 +43,7 @@ pub struct Application {
     compositor: Compositor,
     editor: Editor,
 
-    // TODO: share an ArcSwap with Editor?
-    config: Config,
+    config: Arc<ArcSwap<Config>>,
```

Comment:
This seems wrong, `ArcSwap` already contains an `Arc`. 

Should be `ArcSwap<Config>`

Reply:
This pattern is [described here](https://docs.rs/arc-swap/latest/arc_swap/docs/patterns/index.html#sharing-of-configuration-data), the important bits being:

> The idea here is, each new version is a newly allocated in its own [Arc](https://doc.rust-lang.org/nightly/alloc/sync/struct.Arc.html). It is then stored into a shared ArcSwap instance.

> Note that the configuration needs to be passed through a single shared [ArcSwap](https://docs.rs/arc-swap/latest/arc_swap/type.ArcSwap.html). That means we need to share that instance and we do so through an [Arc](https://doc.rust-lang.org/nightly/alloc/sync/struct.Arc.html) (one could use a global variable instead).

I think for those reasons, we actually do want this right?

Reply:
Hmm, I see 👍🏻